contracts/SportMarkets/SportsAMM.sol

```diff
@@ -820,32 +795,35 @@ contract SportsAMM is Initializable, ProxyOwned, PausableUpgradeable, ProxyReent
         uint amount,
         uint expectedPayout,
         uint additionalSlippage,
-        address collateral
+        address collateral,
+        bool isEth
     ) internal {
-        int128 curveIndex = _mapCollateralToCurveIndex(collateral);
-        require(curveIndex > 0 && curveOnrampEnabled, "unsupported collateral");
-
         (uint collateralQuote, uint susdQuote) = buyFromAmmQuoteWithDifferentCollateral(
             market,
             position,
             amount,
             collateral
         );
 
-        uint transformedCollateralForPegCheck = collateral == usdc || collateral == usdt
-            ? collateralQuote * (1e12)
-            : collateralQuote;
-        require(
-            maxAllowedPegSlippagePercentage > 0 &&
-                transformedCollateralForPegCheck >= (susdQuote * (ONE - (maxAllowedPegSlippagePercentage))) / ONE,
-            "Max peg slippage"
-        );
+        require((collateralQuote * ONE) / (expectedPayout) <= (ONE + additionalSlippage), "Slippage too high!");
+
+        uint exactReceived;
+
+        if (isEth) {
+            require(collateral == multiCollateralOnOffRamp.WETH9(), "Wrong collateral sent");
```

Comment:
the require statement has not clear point here, the call to this line is hardcoded already in  `buyFromAMMWithEthAndReferrer()` with the bool flag, the collateral address is not used in the remaining part.

Removing it releases space and saves gas

Reply:
true, but the quote above uses the collateral variable, so it feels safer to have that require statement, also in case some refactoring is done this method might be called from other public methods.